### ISSUE TITLE
missing vertex and face textures

### ISSUE THREAD

**Issue by akshaygadipatil** [3D-FRONT-FUTURE/3D-FRONT-ToolBox#2]
Hi,

Thanks for the helpful scripts. I saved the obj files, but without vertex and face textures, i.e., without 'vt' and 'ftc'.
'vt' and 'ftc' variables were missing when the execution comes to [Line 138 of run.py](https://github.com/3D-FRONT-FUTURE/3D-FRONT-ToolBox/blob/a2fd284d9d87a1c1757065b7e3a03ea7ac92062b/json2obj/run.py#L138).
Removing these variables from Line 138 helps to save the obj files (without texture info), but is it the case, or am I missing something?

**Comment by jwzxgy2007** [comment]
Hi,
Thanks for your feedback. Could you please describe your system info? I run the script and the obj file has vt and ftc.

**Comment by akshaygadipatil** [comment]
> Hi,
> Thanks for your feedback. I add the missing import lib and the saved obj file has vt and ftc. You can try it again. If it is still wrong, please describe your system info.

Thank you for your quick response.
The error still exists for me. Please see the snapshot below.
![image](https://user-images.githubusercontent.com/22264995/87881467-3cd05480-c9ae-11ea-8c3b-24b8313ac93c.png)

My system info is: Windows 10 64-bit. 
I am not sure what else to provide for the system info, but happy to provide specific details that you may ask. Pls let me know.

**Comment by jwzxgy2007** [comment]
Could you please tell me which python you use, 2 or 3?  and what the version of igl and trimesh lib?

**Comment by akshaygadipatil** [comment]
I am using python 3.7 and the versions of "igl" and "trimesh" are the ones that I downloaded using the commands provided. I think they use the versions available on Anaconda, i.e., "igl" 2.2.0 and "trimesh" 3.7.12.

**Comment by jwzxgy2007** [comment]
My env is same with yours.  I think maybe your 'future_path' param is not valid, and you can check if below path is valid
`args.future_path+'/' + model_jid[idx]`

**Comment by akshaygadipatil** [comment]
Thanks for your help. It works. The problem was with the Pycharm version on Windows, I believe. When I ran it on Linux, everything seemed to work. Still weird.....

**Comment by akshaygadipatil** [comment]
Another question: I don't see any furniture in the obj files obtained after running the json2obj.py script. Am I missing something? How to visualize the scenes with furniture present?

**Comment by gaolinorange** [comment]
Hi, after running the file, if you get these obj files? If not, please check if the `future_path` param is valid and `igl.read_obj` function  is work.
![](https://user-images.githubusercontent.com/5219012/89601098-43a80400-d896-11ea-8382-535e21c487dd.png)

**Comment by jwzxgy2007** [comment]
Hi, after running the py file, if you get obj files like this image? If not, please check if the `future_path` param is valid and `igl.read_obj` function is work.
![](https://user-images.githubusercontent.com/14820455/89601270-badd9800-d896-11ea-83d3-b107db7543b8.png)

**Comment by akshaygadipatil** [comment]
Oh, right. Sorry, yes, I see the objects. But not for all room types (ex: kitchen, bathroom). Are only a few room types supported with furniture?

**Comment by jwzxgy2007** [comment]
Yes, the kitchen and bathroom don't contain furniture.  We will open more ornament and the texture of the floors, walls once these are prepared.
Please check the updated news on the website.

**Comment by ttsesm** [comment]
Hi,

Thanks for the dataset, it seems really promising.

I have a couple of questions as well:

1. When I run the `json2obj.py` script I am getting these warning messages:

```
a826f9e0-54d6-472e-92cc-e9b053526c74
Warning: readOBJ() ignored non-comment line 54:
  o solid_001
Warning: readOBJ() ignored non-comment line 121:
  o solid_002
Warning: readOBJ() ignored non-comment line 263:
  o solid_003
Warning: readOBJ() ignored non-comment line 361:
  o solid_004
Warning: readOBJ() ignored non-comment line 739:
  o solid_005
Warning: readOBJ() ignored non-comment line 1319:
  o solid_006
Warning: readOBJ() ignored non-comment line 2373:
  o solid_007
Warning: readOBJ() ignored non-comment line 4683:
  o solid_008
Warning: readOBJ() ignored non-comment line 5875:
  o solid_009
Warning: readOBJ() ignored non-comment line 1722:
  o solid_001
Warning: readOBJ() ignored non-comment line 14859:
  o solid_002
Warning: readOBJ() ignored non-comment line 39780:
  o solid_003
Warning: readOBJ() ignored non-comment line 50530:
  o solid_004
Warning: readOBJ() ignored non-comment line 57902:
  o solid_005
Warning: readOBJ() ignored non-comment line 62742:
  o solid_006
Warning: readOBJ() ignored non-comment line 1451:
  o solid_001
Warning: readOBJ() ignored non-comment line 2861:
  o solid_002
Warning: readOBJ() ignored non-comment line 1451:
  o solid_001
Warning: readOBJ() ignored non-comment line 2861:
  o solid_002
Warning: readOBJ() ignored non-comment line 1451:
  o solid_001
Warning: readOBJ() ignored non-comment line 2861:
  o solid_002
1c1fca70-5f24-43b2-8a84-1f61b8f2a13a
Warning: readOBJ() ignored non-comment line 23810:
  o solid_001
Warning: readOBJ() ignored non-comment line 40658:
  o solid_002
```
is that something I should be worry about it or it is fine?

2. Is there any progress on adding texture information for the room surfaces? It is really important to have this available ASAP.
3. On some `mesh.obj` files when I load them in meshlab I am getting the following error message:
![image](https://user-images.githubusercontent.com/10189018/91581384-abf67c80-e94e-11ea-8edc-9b18683acb9a.png)
4. In some rooms the faces also seem to be kind of corrupted, check the walls on the left and right side in the following image:
![image](https://user-images.githubusercontent.com/10189018/91581908-65555200-e94f-11ea-9e88-8d421ccb71a9.png)
I do not know if this could be related to 3.
5. If I load all the meshes from the folders of each scene they also seem not to be correctly aligned. E.g. check the screenshot below:
![image](https://user-images.githubusercontent.com/10189018/91582424-152abf80-e950-11ea-98a1-700080ae0a3f.png)
1, 2, 3, and 4 correspond to the balcony, bedroom, kitchen and bathroom respectively from the scene 1c1fca70-5f24-43b2-8a84-1f61b8f2a13a as you can see there is a gap between room 3 and 4 while the lamp in room 2 as you can see goes out of the ceiling. I am not sure whether this should be happening or not.

Thanks.

**Comment by ttsesm** [comment]
Ok,for problem No. 5 it was my mistake I have forgotten to load the living room. Thus, there is not gap between the rooms. Apologies my bad. The lamp though is exceeding the ceiling.

**Comment by jwzxgy2007** [comment]
Hi, thanks for your feedback.
1. It is fine.
2. The texture of surface will be added in the future, but now I can't give you an accurate date.
3.4. We will fix these errors.

**Comment by ttsesm** [comment]
@jwzxgy2007 thank you for the prompt response :-).

Any approximation about when do you believe that the layout textures and these errors could be resolved? Is there any way we could help so that to speed up?

**Comment by ttsesm** [comment]
@jwzxgy2007 any update?

Also I've noticed that the material properties are relying only to the texture images, any possibility  that you could provide other properties as well, e.g. `Ka`, `Kd`, `Ks` etc. and/or even provide scene properties to the Mitsuba scene description file format.

**Comment by jwzxgy2007** [comment]
> @jwzxgy2007 thank you for the prompt response :-).
> 
> Any approximation about when do you believe that the layout textures and these errors could be resolved? Is there any way we could help so that to speed up?

@ttsesm I'm sorry for late reply. I'm afrried that I can't give a exact time for open the texture of walls but we are working on it. The refined 3D-FRONT dataset and Trescope, a light-weight rendering tool, will be released soon. Please check the website for news,
[https://tianchi.aliyun.com/specials/promotion/alibaba-3d-scene-dataset](https://tianchi.aliyun.com/specials/promotion/alibaba-3d-scene-dataset)
thank you!

**Comment by small-zeng** [comment]
> Hi, thanks for your feedback.
> 
> 1. It is fine.
> 2. The texture of surface will be added in the future, but now I can't give you an accurate date.
>    3.4. We will fix these errors.

Has the json2obj.py script been updated for textures? I ran it but still no textures.

**Comment by small-zeng** [comment]
Thanks， but when I run it, it occurs
![image](https://github.com/3D-FRONT-FUTURE/3D-FRONT-ToolBox/assets/63596020/2d532b9a-4d02-4d1f-ab75-0fde3433a4df)

**Comment by thousanddrowned** [comment]
> Thanks， but when I run it, it occurs谢谢，但当我运行它时，出现 ![image](https://github.com/3D-FRONT-FUTURE/3D-FRONT-ToolBox/assets/63596020/2d532b9a-4d02-4d1f-ab75-0fde3433a4df)

Hi, Can json2obj generate textured scenes?
